include/ht_shadervariable.h

```diff
@@ -73,21 +73,21 @@ namespace Hatchit {
         template<>
         inline void HT_API ShaderVariableTemplate<int>::SetData(int data)
```

Comment:
All these template specializations can be re-written as a general template of T.
```C++
template<typename T>
inline void HT_API ShaderVariableTemplate::SetData(T data)
{
    if (m_data != nullptr)
        delete m_data;
    m_data = new T(data);
    type = typeid(T);
}
```

And then member variable m_type could be of type std::type_info